txclib/utils.py

```diff
@@ -195,8 +200,10 @@ def make_request(method, host, url, username, password, fields=None,
                 raise HttpNotAuthorized(data)
             elif response.status == 404:
                 raise HttpNotFound(data)
-            else:
-                raise Exception(data)
+            elif response.status >= 500:
```

Comment:
You are still missing some cases, e.g. 400. You should keep the original `else` clause and add the extra extra elif before that.